src/runtime/virtcontainers/errors/errors.go

```diff
@@ -0,0 +1,117 @@
+package errors
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Interface implmented by pkg/errors
+type stackTracer interface {
+	StackTrace() errors.StackTrace
+}
+
+// Interface implmented by pkg/errors
+type causer interface {
+	Cause() error
+}
+
+// ErrorContext: Helper function for adding a context to an error.
+// err: An error pointer to add the context to.
+// ctx: The context to add to the error.
+// if err is nil, nothing happens.
+func ErrorContext(err *error, ctx string) {
+	if *err == nil {
+		return
+	}
+	if _, ok := (*err).(causer); !ok {
+		*err = errors.New((*err).Error())
+	}
+	*err = errors.Wrap(*err, ctx+"\n\tCause")
+}
+
+// ErrorReport: Helper function to format a high level error.
+// err: An error pointer to format.
+// if err is nil, nothing happens.
+// return a formated error which includes the stack trace and the cause.
+// Format:
+// Cause: <most-inner-error>
+// Error trace:
+// 	0. <most-inner-error>
+// 	1. <...>
+// 	N. <least-inner-error>
+//
+// Stack:
+// 0. <path-to-most-inner-function>
+//  <file-path>: +<line-number>
+// ...
+// Example:
+//
+// Cause: error
+// Error trace:
+// 	0. error
+// 	1. context2
+// 	2. context1
+//
+// Stack:
+// 0. github.com/kata-containers/kata-containers/src/runtime/virtcontainers/errors.AddContext
+// 	/mnt/go/src/github.com/kata-containers/kata-containers/src/runtime/virtcontainers/errors/errors.go +`
+func ErrorReport(err error) error {
+	if err == nil {
+		return err
```

Comment:
Suggestion: Replace `return nil` instead of `return err`, though they are the same result, the first one is more simple to understand.